modules/apps/change-tracking/change-tracking-service/src/main/java/com/liferay/change/tracking/service/impl/CTCollectionLocalServiceImpl.java

```diff
@@ -525,24 +524,6 @@ public void discardCTEntry(
 		Map<Long, Set<Long>> enclosureMap = CTEnclosureUtil.getEnclosureMap(
 			ctClosure, modelClassNameId, modelClassPK);
 
-		for (Map.Entry<Long, Long> entry :
```

Comment:
We have always had children that can't live without their parents.  It has never required this change.  Are you sure this is needed?

Reply:
Well, EmailAddressTableReferenceDefinitionTest fails without this change. Also the tests fail for Address and Phone on my dev branch.

If I understand it correctly, the code I deleted fetches the ctEntry's parent, and tries to make sure that it doesn't exist, otherwise it throws an error.
I'm not sure what was the original intention, but this won't work for entries which are connected to an Organization, because we can delete e.g. an Email Address, without deleting the parent organization.

Reply:
I tried this, with the same result, the test is failing.
https://github.com/moltam89/liferay-portal/pull/new/LPS-161618-childParent

What is the difference between linking form parent to child vs child to parent?

Reply:
from: https://liferay.atlassian.net/wiki/spaces/ENGSTAGING/pages/2084896832/How+To+Implement+a+TableReferenceDefinition+for+an+Entity+that+is+going+to+be+change+tracked

Good way to tell what is parent/child is that the parent creates the children when it is created (like with assets & resources)
Defining as parent means we do a missing parent conflict check and allow discarding the children without discarding the parent
Defining as a child means we don't check for missing parent and we do not allow discarding without cascading up to the parent

Reply:
Do you happen to know where is this distinction in the code?
Because the code part I deleted doesn't seem to care whether I define the connection from parent to child, or child to parent.